discord_slash/client.py

```diff
@@ -717,39 +724,49 @@ async def on_socket_response(self, msg):
 
         to_use = msg["d"]
 
-        if to_use["data"]["name"] in self.commands:
-
-            ctx = context.SlashContext(self.req, to_use, self._discord, self.logger)
-            cmd_name = to_use["data"]["name"]
+        ctx = context.SlashContext(self.req, to_use, self._discord, self.logger)
+        cmd_name = to_use["data"]["name"]
 
-            if cmd_name not in self.commands and cmd_name in self.subcommands:
+        # check guild commands and global commands for this command
+        if ctx.guild_id in self.commands and \
+                to_use["data"]["name"] in self.commands[ctx.guild_id]:
+            selected_cmd = self.commands[ctx.guild_id][to_use["data"]["name"]]
+        elif to_use["data"]["name"] in self.commands["global"]:
+            selected_cmd = self.commands["global"][to_use["data"]["name"]]
```

Comment:
pretty sure that this'll select a guild command to run if it exists, even if a global command was called
because `ctx.guild_id` is only none in dms
I've had a look at the [docs](https://discord.com/developers/docs/interactions/slash-commands#interaction) regarding the response received from discord, and it looks like they don't tell you if it's global or guild though